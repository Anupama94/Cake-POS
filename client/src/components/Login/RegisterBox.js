import React from 'react';
import './Login.css';
import { Form, FormGroup, Label, Input, FormFeedback, FormText, Col } from 'reactstrap';


export class RegisterBox extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            'username': '',
            'email': '',
            'password': '',
            validate: {
                usernameState: '',
                emailState: '',
                passwordState: ''
            },
        }
        this.handleChange = this.handleChange.bind(this);
    }


    validateEmail(e) {
        const emailRex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
        const { validate } = this.state;
        if (emailRex.test(e.target.value)) {
            validate.emailState = 'has-success'
        } else {
            validate.emailState = 'has-danger'
        }
        this.setState({ validate })
    }


    validatePassword(e) {
        const passwordRex = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{4,}$/;
        const { validate } = this.state;
        if (passwordRex.test(e.target.value)) {
            validate.passwordState = 'has-success'
        } else {
            validate.passwordState = 'has-danger'
        }
        this.setState({ validate })
    }

    handleChange = async (event) => {
        const { target } = event;
        const value = target.value;
        const { name } = target;
        await this.setState({
            [name]: value,
        });
    }

    validateUsername(e) {
        /* validate against the database */

    }
    submitRegister(e) {
        e.preventDefault();
        if (this.state.validate.emailState === 'has-success') {
            alert();
        }
    }


    render() {

        const { email, password, } = this.state;

        return (
            <div className="inner-container">
                <div className="header">
                    Register
                </div>
                <div className="box">

                    <Form>
                        <Col>
                            <FormGroup>
                                <Label>Username</Label>
                                <Input
                                    type="text"
                                    name="username"
                                    id="username"
                                    value={email}
                                    valid={this.state.validate.usernameState === 'has-success'}
                                    invalid={this.state.validate.usernameState === 'has-danger'}
                                    onChange={(e) => {
                                        this.validateUsername(e)
                                        this.handleChange(e)
                                    }}
                                />

                                <FormFeedback>
                                    Uh oh! This username is taken!
                                 </FormFeedback>
                            </FormGroup>
                        </Col>

                        <Col>
                            <FormGroup>
                                <Label>Email</Label>
                                <Input
                                    type="email"
                                    name="email"
                                    id="exampleEmail"
                                    placeholder="myemail@email.com"
                                    value={email}
                                    valid={this.state.validate.emailState === 'has-success'}
                                    invalid={this.state.validate.emailState === 'has-danger'}
                                    onChange={(e) => {
                                        this.validateEmail(e)
                                        this.handleChange(e)
                                    }}
                                />

                                <FormFeedback>
                                    Uh oh! Looks like there is an issue with your email. Please input a correct email.
                                </FormFeedback>
                            </FormGroup>
                        </Col>
                        <Col>
                            <FormGroup>
                                <Label for="examplePassword">Password</Label>
                                <Input
                                    type="password"
                                    name="password"
                                    id="examplePassword"
                                    placeholder="Password"
                                    value={password}
                                    onChange={(e) => {
                                        this.validatePassword(e)
                                        this.handleChange(e)
                                    }}
                                />
                                <FormFeedback invalid>
                                    Uh oh! Looks like there is an issue with your email. Please input a correct email.
                                </FormFeedback>
                                <FormText>Password should contain at least four characters, at least one number and both lower and uppercase letters and special characters.</FormText>
                            </FormGroup>
                        </Col>
                    </Form>

                    <button type="button"
                        className="login-btn"
                        onClick={this.submitRegister.bind(this)}> Register
                    </button>
                </div>
            </div>

        );
    }
}

export default RegisterBox;


