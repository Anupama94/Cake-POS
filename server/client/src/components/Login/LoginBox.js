import React from 'react';
import './Login.css';
import { Form, FormGroup, Label, Input, FormFeedback, Col, Alert } from 'reactstrap';
import PropTypes from 'prop-types';
import auth from '../../authentication/auth';
import { userLogin } from '../../apiCalls/callApi';



export class LoginBox extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      validate: {
        emailState: '',
        passwordState: ''
      },
      errorAlert: {
        visible: false,
        message: ''
      }
    }
    this.handleChange = this.handleChange.bind(this);
    this.validateEmail = this.validateEmail.bind(this);
    this.validatePassword = this.validatePassword.bind(this);
  }

  static contextTypes = {
    router: PropTypes.object
  }


  submitLogin = () => {
    
    if ((this.state.validate.emailState !== 'has-success') && (this.state.validate.passwordState !== 'has-success')) {
      this.setState({ validate: { emailState: 'has-danger', passwordState: this.state.validate.passwordState } });
    }

    else if ((this.state.validate.emailState === 'has-success') && (this.state.validate.passwordState !== 'has-success')) {
      this.setState({ validate: { emailState: this.state.validate.emailState, passwordState: 'has-danger' } });
    }
    else if ((this.state.validate.emailState !== 'has-success') && (this.state.validate.passwordState === 'has-success')) {
      this.setState({ validate: { emailState: 'has-danger', passwordState: this.state.validate.passwordState } });
    }
    else {
      userLogin({ email: this.state.email, password: this.state.password })
        .then((response) => {
          auth.login(() => {
            this.context.router.history.push("/OrderList");
          })
        })
        .catch(err => {
          this.setState({ email: '', password: '' });
          this.setState({ validate: { emailState: 'has-danger' } });
          this.setState({ errorAlert: { visible: true, message: err.message } });
        });
    }


  }

  validateEmail(e) {
    const emailRex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

    if (emailRex.test(e.target.value)) {
      this.setState({ validate: { emailState: 'has-success', passwordState: this.state.validate.passwordState } });
    } else {
      this.setState({ validate: { emailState: 'has-danger', passwordState: this.state.validate.passwordState } });
    }

  }


  validatePassword(e) {
    if (e.target.value) {
      this.setState({ validate: { emailState: this.state.validate.emailState, passwordState: 'has-success' } });
    }
    else {
      this.setState({ validate: { emailState: this.state.validate.emailState, passwordState: 'has-danger' } });
    }
  }


  handleChange = async (event) => {
    const { target } = event;
    const value = target.value;
    const { name } = target;
    await this.setState({
      [name]: value,
    });
  }


  render() {
    const { email, password } = this.state;

    return (
      <div className="inner-container">
        <div className="header">
          Log In
          </div>
        <div className="box">

          <Form>
            <Col>
              <FormGroup>
                <Label>Username</Label>
                <Input
                  type="email"
                  name="email"
                  id="exampleEmail"
                  placeholder="myemail@email.com"
                  value={email}
                  valid={this.state.validate.emailState === 'has-success'}
                  invalid={this.state.validate.emailState === 'has-danger'}
                  onChange={(e) => {
                    this.validateEmail(e)
                    this.handleChange(e)
                  }}
                />

                <FormFeedback invalid>
                  Uh oh! Looks like there is an issue with your email. Please input a correct email.
                </FormFeedback>
              </FormGroup>
            </Col>
            <Col>
              <FormGroup>
                <Label for="examplePassword">Password</Label>
                <Input
                  required
                  type="password"
                  name="password"
                  id="examplePassword"
                  placeholder="Password"
                  value={password}
                  valid={this.state.validate.passwordState === 'has-success'}
                  invalid={this.state.validate.passwordState === 'has-danger'}
                  onChange={(e) => {
                    this.validatePassword(e)
                    this.handleChange(e)
                  }
                  }
                />
                <FormFeedback invalid>
                  *Password is required!
                </FormFeedback>
              </FormGroup>
            </Col>
          </Form>

          <Alert color="danger" isOpen={this.state.errorAlert.visible}>
            {this.state.errorAlert.message}
          </Alert>

          <button type="button" className="login-btn" onClick={this.submitLogin}> Login </button>
        </div>
      </div>

    );
  }
}

export default LoginBox;