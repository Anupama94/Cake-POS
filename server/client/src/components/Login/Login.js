import React from 'react';
import './Login.css';
import { Form, FormGroup, Label, Input, FormFeedback, FormText, Col } from 'reactstrap';
import { LoginBox } from './LoginBox';


class LoginOrRegister extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      isLoginOpen: true,
      isRegisterOpen: false,
    };
  }

  showLoginBox() {
    this.setState({ isLoginOpen: true, isRegisterOpen: false });
  }

  showRegisterBox() {
    this.setState({ isRegisterOpen: true, isLoginOpen: false });
  }

  render() {
    return (
      <div className="root-container">
        <div className="box-controller">
          <div className={"controller " + (this.state.isLoginOpen ? "selected-controller" : "")}
            onClick={this.showLoginBox.bind(this)}>
            Log In
          </div>
          <div className={"controller " + (this.state.isRegisterOpen ? "selected-controller" : "")}
            onClick={this.showRegisterBox.bind(this)}>
            Register
          </div>
        </div>
        <div className="box-container">
          {this.state.isLoginOpen && <LoginBox />}
          {this.state.isRegisterOpen && <RegisterBox />}
        </div>
      </div>

    );
  }
}




class RegisterBox extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      'username': '',
      'email': '',
      'password': '',
      validate: {
        usernameState: '',
        emailState: '',
        passwordState: ''
      },
    }
    this.handleChange = this.handleChange.bind(this);
  }


  validateEmail(e) {
    const emailRex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    const { validate } = this.state;
    if (emailRex.test(e.target.value)) {
      validate.emailState = 'has-success'
    } else {
      validate.emailState = 'has-danger'
    }
    this.setState({ validate })
  }


  validatePassword(e) {
    const passwordRex = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{4,}$/;
    const { validate } = this.state;
    if (passwordRex.test(e.target.value)) {
      validate.passwordState = 'has-success'
    } else {
      validate.passwordState = 'has-danger'
    }
    this.setState({ validate })
  }

  handleChange = async (event) => {
    const { target } = event;
    const value = target.value;
    const { name } = target;
    await this.setState({
      [name]: value,
    });
  }

  validateUsername(e) {
    /* validate against the database */

  }
  submitRegister(e) {
    e.preventDefault();
    if (this.state.validate.emailState === 'has-success') {
      alert();
    }
  }


  render() {

    const { username, email, password, } = this.state;

    return (
      <div className="inner-container">
        <div className="header">
          Register
        </div>
        <div className="box">

          <Form>
            <Col>
              <FormGroup>
                <Label>Username</Label>
                <Input
                  type="text"
                  name="username"
                  id="username"
                  value={email}
                  valid={this.state.validate.usernameState === 'has-success'}
                  invalid={this.state.validate.usernameState === 'has-danger'}
                  onChange={(e) => {
                    this.validateUsername(e)
                    this.handleChange(e)
                  }}
                />

                <FormFeedback>
                  Uh oh! This username is taken!
                </FormFeedback>
              </FormGroup>
            </Col>

            <Col>
              <FormGroup>
                <Label>Email</Label>
                <Input
                  type="email"
                  name="email"
                  id="exampleEmail"
                  placeholder="myemail@email.com"
                  value={email}
                  valid={this.state.validate.emailState === 'has-success'}
                  invalid={this.state.validate.emailState === 'has-danger'}
                  onChange={(e) => {
                    this.validateEmail(e)
                    this.handleChange(e)
                  }}
                />

                <FormFeedback>
                  Uh oh! Looks like there is an issue with your email. Please input a correct email.
              </FormFeedback>
              </FormGroup>
            </Col>
            <Col>
              <FormGroup>
                <Label for="examplePassword">Password</Label>
                <Input
                  type="password"
                  name="password"
                  id="examplePassword"
                  placeholder="Password"
                  value={password}
                  onChange={(e) => {
                    this.validatePassword(e)
                    this.handleChange(e)
                  }}
                />
                <FormFeedback invalid>
                  Uh oh! Looks like there is an issue with your email. Please input a correct email.
              </FormFeedback>
                <FormText>Password should contain at least four characters, at least one number and both lower and uppercase letters and special characters.</FormText>
              </FormGroup>
            </Col>
          </Form>

          <button type="button"
            className="login-btn"
            onClick={this.submitRegister.bind(this)}> Register
           </button>
        </div>
      </div>

    );
  }
}


class Login extends React.Component {

  render() {
    return (
      <LoginOrRegister></LoginOrRegister>

    );
  }
}

export default Login;