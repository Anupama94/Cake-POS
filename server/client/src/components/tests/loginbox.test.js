import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Login from '../Login/Login'
import { shallowToJson } from 'enzyme-to-json';


describe("LoginBox Component", () => {

    let wrapper;
    let initialState = {
        email: '',
        password: '',
        validate: {
          emailState: '',
          passwordState: ''
        },
        errorAlert: {
          visible: false,
          message: ''
        }
      };

    beforeEach(() => {
         wrapper = shallow(<Login />); 
        }
    );

    it("should render correctly", () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });


    it('when submit button is clicked without username nor password emailState should be set to has-danger', () => {
        //const submitButton = wrapper.find('button.login-btn').toBeTruthy;
        //expect(foodlist.text()).to.equal("mock menu item name").toBeTruthy;
        //expect(wrapper.find('h2.orangify').text()).toBe("Cake | POS");
    });

   

});
