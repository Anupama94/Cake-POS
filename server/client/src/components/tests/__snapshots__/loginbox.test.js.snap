// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginBox Component should render correctly 1`] = `
<div
  className="inner-container"
>
  <div
    className="header"
  >
    Log In
  </div>
  <div
    className="box"
  >
    <Form
      tag="form"
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <FormGroup
          tag="div"
        >
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Username
          </Label>
          <Input
            id="exampleEmail"
            invalid={false}
            name="email"
            onChange={[Function]}
            placeholder="myemail@email.com"
            type="email"
            valid={false}
            value=""
          />
          <FormFeedback
            invalid={true}
            tag="div"
          >
            Uh oh! Looks like there is an issue with your email. Please input a correct email.
          </FormFeedback>
        </FormGroup>
      </Col>
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <FormGroup
          tag="div"
        >
          <Label
            for="examplePassword"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Password
          </Label>
          <Input
            id="examplePassword"
            invalid={false}
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            valid={false}
            value=""
          />
          <FormFeedback
            invalid={true}
            tag="div"
          >
            *Password is required!
          </FormFeedback>
        </FormGroup>
      </Col>
    </Form>
    <Alert
      closeAriaLabel="Close"
      color="danger"
      fade={true}
      isOpen={false}
      tag="div"
      transition={
        Object {
          "appear": true,
          "baseClass": "fade",
          "baseClassActive": "show",
          "enter": true,
          "exit": true,
          "in": true,
          "mountOnEnter": false,
          "onEnter": [Function],
          "onEntered": [Function],
          "onEntering": [Function],
          "onExit": [Function],
          "onExited": [Function],
          "onExiting": [Function],
          "tag": "div",
          "timeout": 150,
          "unmountOnExit": true,
        }
      }
    />
    <button
      className="login-btn"
      onClick={[Function]}
      type="button"
    >
       Login 
    </button>
  </div>
</div>
`;
